graspy/embed/mds.py

```diff
@@ -51,63 +60,152 @@ class ClassicalMDS(BaseEstimator):
         Equals the parameter n_components, or n_features if n_components
         is None.
 
+    dissimilarity_matrix_ : array, shape (n_features, n_features)
+        Dissimilarity matrix 
+
     See Also
     --------
     graphstats.embed.dimselect
     """
 
-    def __init__(self, n_components):
+    def __init__(self, n_components=None, dissimilarity='euclidean'):
+        # Check inputs
+        if n_components is not None:
+            if not isinstance(n_components, int):
+                msg = "n_components must be an integer, not {}.".format(
+                    type(n_components))
+                raise TypeError(msg)
+            elif n_components <= 0:
+                msg = "n_components must be >= 1 or None."
+                raise ValueError(msg)
         self.n_components = n_components
 
-    def fit(self, X):
+        if dissimilarity not in ['euclidean', 'precomputed']:
+            msg = "Dissimilarity measure must be either 'euclidean' or 'precomputed'."
+            raise ValueError(msg)
+        self.dissimilarity = dissimilarity
+
+    def _compute_euclidean_distances(self, X, n_elements):
+        """
+        Computes pairwise distance between row vectors or matrices
+
+        Parameters
+        ----------
+        X : array, shape (n_samples, n_features) or (n_samples, n_samples)
+            Input data. If ``dissimilarity=='precomputed'``, the input should
+            be the dissimilarity matrix.
+        
+        n_elements : int, optional, default: 1
+            If n_elements > 1, it computes the pairwise euclidean distance between
+            matrices given by (n_elements, n_features). n_elements must divide evenly 
```

Comment:
about the n_elements explanation - when you say the matrices are given by (n_elements, n_features) do you mean that is the dimension of each resulting matrix that it compares?

Reply:
Yea. So if n_elements=5, and your input X is 100 by 2 matrix, then it takes the first 5 rows as one sample matrix and computes distance to the second 5 rows, etc. Still not sure how to word this clearly.

Reply:
Another approach is to allow for any n-dimensional arrays and compute distance along axis=0. So for outputs of omni, reshape it such that it has shape (n_graphs, n_vertices, embedded_dimensions).